appgate/files.py

```diff
@@ -0,0 +1,144 @@
+        file_url = f"{address}/{file_key}"
+
+        async def get(url: str) -> str:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url) as response:
+                    return base64.b64encode(await response.read()).decode()
+
+        return asyncio.run(get(file_url))
+
+
+class AppgateS3File(AppgateFile):
+    def load_file(self) -> str:
+        address = os.getenv("APPGATE_FILE_S3_ADDRESS", "localhost")
+        access_key = os.getenv("APPGATE_S3_ACCESS_KEY")
+        secret_key = os.getenv("APPGATE_S3_SECRET_KEY")
+        secure = os.getenv("APPGATE_S3_SSL_NO_VERIFY") == "false"
+
+        client = Minio(
+            endpoint=address,
+            access_key=access_key,
+            secret_key=secret_key,
+            secure=secure,
+        )
+        bucket = "sdp"
+        object_key = f"{self.entity_name.lower()}-{self.api_version}/{self.value.get('filename')}"
+
+        if not client.bucket_exists(bucket):
+            raise AppgateFileException("Bucket sdp does not exist on %s", address)
+
+        try:
+            response = client.get_object(bucket, object_key)
+            try:
+                return base64.b64encode(response.data).decode()
+            finally:
+                response.close()
+                response.release_conn()
+        except Exception as e:
+            raise AppgateFileException(
+                "Unable to fetch the file contents for %s: %s", e
+            )
+
+
+def get_appgate_file(value: Dict, entity_name: str) -> AppgateFile:
+    match os.getenv("APPGATE_FILE_SOURCE", ""):
+        case "http":
+            return AppgateHttpFile(value, entity_name)
+        case "s3":
+            return AppgateS3File(value, entity_name)
+        case _:
+            raise AppgateFileException("Unable to create an AppgateFile")
+
+
+def appgate_file_load(value: OpenApiDict, entity_name: str) -> str:
+    appgate_file = get_appgate_file(value, entity_name)
+    return appgate_file.load_file()
+
+
+class FileAttribMaker(AttribMaker):
+    def values(
+        self,
+        attributes: Dict[str, "AttribMaker"],
+        required_fields: List[str],
+        instance_maker_config: EntityClassGeneratorConfig,
+    ) -> AttributesDict:
+        values = super().values(attributes, required_fields, instance_maker_config)
+        if "metadata" not in values:
+            values["metadata"] = {}
+        values["eq"] = (
+            "writeOnly" in values["metadata"] and not values["metadata"]["writeOnly"]
+        )
+        values["metadata"][K8S_LOADERS_FIELD_NAME] = [
+            FileAttribLoader(
+                loader=lambda v: appgate_file_load(
+                    v, instance_maker_config.entity_name
+                ),
+                field=self.name,
+                load_external="APPGATE_FILE_SOURCE" in os.environ,
+            )
+        ]
+        return values
+
+
+def file_attrib_maker(
```

Comment:
mmm do we need a wrapper here?

Reply:
Ah yeah. This is literally just passing through one stack of method call haha 😄

Reply:
Got thrown off by the `checksum_attrib_maker` and `size_attrib_maker` but those makes sense to have a wrapper